src/issuance.rs

```diff
@@ -288,25 +336,27 @@ impl IssueBundle<Unauthorized> {
             &mut rng,
         );
 
-        match self
+        let found = match self
             .actions
             .iter_mut()
             .find(|issue_action| issue_action.asset_desc.eq(&asset_desc))
         {
             // Append to an existing IssueAction.
             Some(action) => {
-                if action.finalize {
-                    return Err(IssueActionAlreadyFinalized);
-                };
                 action.notes.push(note);
-                finalize.then(|| action.finalize = true);
-            }
-            // Insert a new IssueAction.
-            None => {
-                let mut action = IssueAction::new(asset_desc, &note);
-                finalize.then(|| action.finalize = true);
-                self.actions.push(action);
+                true
             }
+            None => false,
+        };
+
+        // Insert a new IssueAction.
+        if !found {
+            let action = IssueAction {
+                asset_desc,
+                notes: vec![note],
+                finalize: false,
+            };
+            self.actions.push(action);
```

Comment:
let's move this code into `None => false,` and return from there

Reply:
Done